src/main/java/pl/mmieczak/springbootexercise_22_1/UserRepository.java

```diff
@@ -22,4 +24,40 @@ public UserRepository() {
     public void add(String name, String surname, int age) {
         userCollection.add(new User(name, surname, age));
     }
+
+    public void removeAll(Stream<User> streamToRemove) {
+        userCollection.removeAll(streamToRemove.collect(Collectors.toList()));
+    }
+
+    public Stream<User> filterDataUsingFormValues(String name, String surname, String age){
```

Comment:
Odpowiadając na pytanie dało się to zrobić prościej. Zauważ że to co my chcieliśmy tutaj osiągnąć to sprawdzenie czy dany użytkownik pasuje do wpisanych "filtrów". Wiemy że każdy parametr jest opcjonalny, więc możemy założyć że jeżeli ktoś czegoś nie wpisał to nie musimy nakładać kolejnych "obostrzeń" jeśli chodzi o użytkowników. Tak więc każdy użytkownik z danym atrybutem spełnia oczekiwania kogoś kto z tego korzysta. Wystarczy się więc po wszystkich użytkownikach przeiterować i sprawdzić czy dany użytkownik ma takie same atrybuty które zostały podane przez kogoś korzystającego z naszego systemu. Sprawdzamy więc dla każdego użytkownika czy dany atrybut został wpisany (czy coś jest `null` bądź puste) oraz czy pasuje on do tego co ma w sobie użytkownik. I tak dla każðego atrybutu podanego przez użytkownika. W kodzie wyglądałoby to następująco:
```
private boolean checkSearchConditionsForUser(String imie, String nazwisko, Integer wiek, User user) {
        boolean nameOk = true;
        boolean surnameOk = true;
        boolean ageOk = true;

        if(!StringUtils.isEmpty(imie) && !imie.equals(user.getName())) {
            nameOk = false;
        }

        if(!StringUtils.isEmpty(nazwisko) && !nazwisko.equals(user.getLastName())) {
            surnameOk = false;
        }

        if(wiek != null && wiek != user.getAge()) {
            ageOk = false;
        }

        return nameOk && surnameOk && ageOk;
    }
```

Reply:
Faktycznie :) proste i przejrzyste. Dzieki!